Two compiler warnings

```
  CC [M]  /g/g24/rountree/v/tioga/repos/msr-safe/msr_entry.o
/g/g24/rountree/v/tioga/repos/msr-safe/msr_entry.c: In function ‘msr_open’:
/g/g24/rountree/v/tioga/repos/msr-safe/msr_entry.c:181:25: warning: unused variable ‘c’ [-Wunused-variable]
  181 |     struct cpuinfo_x86 *c;
      |                         ^
```

and

```
cc  -DVERSION=\"1.6.0\" -fPIC -shared -c msrsave/msrsave_main.c -o msrsave/msrsave_main.o
clang-15: warning: argument unused during compilation: '-shared' [-Wunused-command-line-argument]
cc  -DVERSION=\"1.6.0\" -fPIC -shared -c msrsave/msrsave.c -o msrsave/msrsave.o
clang-15: warning: argument unused during compilation: '-shared' [-Wunused-command-line-argument]
```

--- a/Makefile
+++ b/Makefile
@@ -29,10 +29,10 @@ check: msrsave/msrsave_test
 CURRENT_VERSION := -DVERSION=\"1.6.0\"
 
 msrsave/msrsave.o: msrsave/msrsave.c msrsave/msrsave.h
-	$(CC) $(CFLAGS) $(CURRENT_VERSION) -fPIC -shared -c msrsave/msrsave.c -o $@
+	$(CC) $(CFLAGS) $(CURRENT_VERSION) -fPIC -c msrsave/msrsave.c -o $@
 
 msrsave/msrsave_main.o: msrsave/msrsave_main.c msrsave/msrsave.h
-	$(CC) $(CFLAGS) $(CURRENT_VERSION) -fPIC -shared -c msrsave/msrsave_main.c -o $@
+	$(CC) $(CFLAGS) $(CURRENT_VERSION) -fPIC -c msrsave/msrsave_main.c -o $@
 
 msrsave/msrsave: msrsave/msrsave_main.o msrsave/msrsave.o
 	$(CC) $(CFLAGS) $(CURRENT_VERSION) $^ -o $@

--- a/msr_entry.c
+++ b/msr_entry.c
@@ -178,7 +178,6 @@ static ssize_t msr_write(struct file *file, const char __user *buf, size_t count
 static int msr_open(struct inode *inode, struct file *file)
 {
     unsigned int cpu = iminor(file->f_path.dentry->d_inode);
-    struct cpuinfo_x86 *c;
 
     if (cpu >= nr_cpu_ids || !cpu_online(cpu))
     {

